Update ServletMapping for Async dispatch

pmd1nh opened the pull request:
https://github.com/openliberty/open-liberty/issues/22971

pmd1nh committed:
Update ServletMapping for Async dispatch

pmd1nh committed:
FAT servlet mapping for Async dispatch

pmd1nh commented:
#build
       #spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

pmd1nh committed:
Update Servlet 4.0 for trace and repeat tests in EE10

pmd1nh commented:
Added traces to Servlet 4.0 HttpServletMappingImpl.
Updated servlet.4.0_fat servletMapping async tests to verify the new behaviors when repeat in EE10.

#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

pmd1nh commented:
added a comment to test, no need for new PB.

pnicolucci commented on dev/io.openliberty.webcontainer.servlet.6.0.internal_fat/test-applications/GetMappingAsyncDispatchTest.war/src/servlets/ServletMapping.java:
```diff
@@ -0,0 +1,118 @@
+        response.setHeader("TestResult", sBuilderResponse.toString());  
+        LOG.info("Test ServletMapping dispatched Done.");
+    }
+} 
```
Just add a new line here

pmd1nh commented:
#libby

pmd1nh commented:
rebased and updated comments.

#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.0.internal_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.session_fat,com.ibm.ws.session_fat_lWAS_servlet.3.1.toleration,com.ibm.ws.session.cache_fat,com.ibm.ws.session.cache_fat_config,com.ibm.ws.session.cache_fat_config_infinispan,com.ibm.ws.session.cache_fat_infinispan,com.ibm.ws.session.cache_fat_infinispan_container,com.ibm.ws.session.cache_provider_fat

pmd1nh commented:
#libby
